packages/core/getUserAvailability.ts

```diff
@@ -495,10 +496,44 @@ const _getBusyTimesFromLimits = async (
     performance.measure(`checking duration limits took $1'`, "durationLimitsStart", "durationLimitsEnd");
   }
 
+  const busyTimes = limitManager.getBusyTimes().sort((a, b) => (dayjs(a.start).isBefore(b.start) ? -1 : 1));
+
+  if (eventType.seatsPerTimeSlot) {
+    const bookingsWithRemainingSeats = bookings.filter(
+      (booking) =>
+        eventType.seatsPerTimeSlot &&
+        booking.attendeesCount &&
+        booking.attendeesCount < eventType.seatsPerTimeSlot &&
+        dayjs(booking.start).isBetween(dateFrom, dateTo)
+    );
+
+    for (let i = 0; i < bookingsWithRemainingSeats.length; i++) {
+      for (let j = i; j < busyTimes.length; j++) {
+        const booking = bookingsWithRemainingSeats[i];
+        const busyTime = busyTimes[j];
+        if (dayjs(booking.start).add(1, "ms").isBetween(busyTime.start, busyTime.end)) {
+          busyTimes.splice(
+            j,
+            1,
+            {
+              start: busyTime.start,
+              end: dayjs(booking.start).subtract(1, "ms").toISOString(),
+            },
+            {
+              start: dayjs.utc(booking.end).toISOString(),
+              end: busyTime.end,
+            }
+          );
+          break;
+        }
+      }
+    }
+  }
```

Comment:
**Non-blocking**: Something about a loop inside of a loop didn't sit well with me. What if instead of iterating through `bookings` and then looping through that new array. What if we iterate through bookings, see if it fits the criteria for `bookingWithRemainingSeats`, then splice `busyTimes`. If it doesn't fit the criteria continue to the next booking.

Reply:
I've gone ahead and made this change in this commit b5a43cce44ffc4628b5d947761de182a9b57cc2b